src/js/kits/model-kit-select/model-kit-select.js

```diff
@@ -0,0 +1,152 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import { ModelKitConsumer } from '../../data/kits/modelkit';
+import { BrandSelect } from './brand-select';
+import { ScaleSelect } from './scale-select';
+import { SearchInput } from './search-input';
+import { KitPreview } from './kit-preview';
+
+const DEBOUNCE = 300;  // debounce in ms
+
+const modelKitConsumer = new ModelKitConsumer();
+
+// TODO: pass selected kit IDs as prop
+// TODO: handle toggle changes & allowMultiple yes/no
+
+const ModelKitSelect = (props) => {
+    const {
+        label,
+        htmlName,
+        allowMultiple,
+        selected,
+    } = props;
+
+    // track search parameters state
+    const [searchParams, setSearchParams] = useState({});
+    const emptySearch = !Object.keys(searchParams).length;
+    // wrap around new object creation to update the search params, as key modification
+    // is not observed by React. Also clears the search key if the value is empty.
+    const updateSearchParam = (param, value) => {
+        const newSearchParams = Object.assign({}, searchParams, {[param]: value});
+        if (!value) {
+            delete newSearchParams[param];
+        }
+
+        if (param !== 'page') {
+            delete newSearchParams.page;
+            setSearchResults([]);
+        }
+
+        setSearchParams(newSearchParams);
+    }
+
+    const [searchResults, setSearchResults] = useState([]);
+    const [lastResultList, setLastResultList] = useState(null);
+
+    // make an API call whenever the search params change
+    useEffect(() => {
+        if (emptySearch) {
+            return;
+        }
+        const tid = window.setTimeout(
+            () => {
+                modelKitConsumer
+                    .filter(searchParams)
+                    .then(resultList => {
+                        const newSearchResults = searchResults.concat(resultList);
+                        setSearchResults(newSearchResults);
+                        setLastResultList(resultList);
+                    })
+                    .catch(console.error);
+            }, DEBOUNCE
+        );
+        return () => window.clearTimeout(tid);
+    }, [searchParams]);
+
+    const nextPage = lastResultList ? parseInt(lastResultList.page || 1, 10) + 1 : null;
```

Comment:
I think it's better to use nullish coalescing here `lastResultList.page ?? 1`, otherwise I think if the `lastResultList.page` is `0`, it will return `1` since `0` is falsy. `??` explicitly checks for `null` and `undefined`.

Reply:
page 0 is not possible though, counting starts at 1. But still better to be safe than sorry.